src/resource/worker-control.c

```diff
@@ -35,6 +36,8 @@
 
 #define DEFAULT_WORKER_IDLE_TIMEOUT_USEC 5000000
 
+static int sid_parent_exited = 0;
```

Comment:
Just realized this - we should add this flag somewhere in the `struct worker` directly, because we can have more than one instance of `worker-control` and then its instances would interfere with each other through this global variable.

Reply:
@prajnoha do we have access to the struct worker in the _on_worker_signal_event()?

I'm stepping through the structures in the debugger and it isn't obvious to me what you asking.

Reply:
I thought we are already passing the worker resource to the signal handler, because it's already used there:

https://github.com/sid-project/sid/blob/93563cd39dfd686028da274e3f972f1660b5dba0/src/resource/worker-control.c#L1083-L1088

But then I checked how this is actually set up and noticed there's a tiny bug, should be fixed with this: https://github.com/prajnoha/sid/commits/fix_resource_signal_handler_data